src/main/java/com/minecolonies/util/StructureWrapper.java

```diff
@@ -705,40 +626,44 @@ public TileEntity getTileEntity()
         {
             return null;
         }
-        return this.schematicWorld.getTileEntity(this.progressPos);
+        return this.structure.getTileEntity(this.progressPos);
     }
 
     /**
-     * @return A list of all the entities in the schematic.
+     * @return A list of all the entities in the structure.
      */
     @NotNull
     public List<Entity> getEntities()
     {
-        return schematicWorld.getEntities();
+        return structure.getEntities();
     }
 
     /**
-     * Base position of the schematic.
+     * Base position of the structure.
      *
-     * @return BlockPos representing where the schematic is.
+     * @return BlockPos representing where the structure is.
      */
     public BlockPos getPosition()
     {
+        if(position == null)
+        {
+            return new BlockPos(0,0,0);
+        }
         return position;
     }
 
     /**
      * Set the position, used when loading.
      *
-     * @param position Where the schematic is in the world.
+     * @param position Where the structure is in the world.
      */
     public void setPosition(BlockPos position)
     {
         this.position = position;
     }
 
     /**
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) First sentence should end with a period. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=checkstyle%3AJavadoc_Style)